File: nodejs/geo-targeting.js
const { chromium } = require("playwright");

const proxy = {
    server: "http://resi.enigmaproxy.net:12321",

    // Change this to target different locations
    username: "USERNAME_country-us",

    password: "PASSWORD"
};

(async () => {

    const browser = await chromium.launch({
        headless: true,
        proxy
    });

    try {

        const page = await browser.newPage();

        console.log("Checking location...");

        await page.goto("https://httpbin.org/ip", {
            waitUntil: "networkidle"
        });

        console.log(await page.locator("body").textContent());

    } finally {

        await browser.close();

    }

})();

File: python/geo_targeting.py
from playwright.sync_api import sync_playwright

proxy = {
    "server": "http://resi.enigmaproxy.net:12321",

    # Change this to target different locations
    "username": "USERNAME_country-us",

    "password": "PASSWORD"
}

with sync_playwright() as p:

    browser = p.chromium.launch(
        headless=True,
        proxy=proxy
    )

    try:

        page = browser.new_page()

        print("Checking location...")

        page.goto(
            "https://httpbin.org/ip",
            wait_until="networkidle"
        )

        print(page.locator("body").text_content())

    finally:

        browser.close()